_posts/2021-07-25-personal-allowance.md

```diff
@@ -0,0 +1,1319 @@
+    x.style.display = "block";
+  } else {
+    x.style.display = "none";
+  }
+}
+</script>
+
+
+<button onclick="show_code_7()">Click to show code</button>
+<div id="code_block_7" style="display: none;">
+  <pre>
+    <code>
+sim_BI = sim_list[0]
+PA_amounts = [12500, 11500, 10500, 9500, 8500, 7500, 6500, 5500, 4500, 3500, 2500, 1500, 500, 0]
+PA_amounts2 = [0, 500, 1500, 2500, 3500, 4500, 5500, 6500, 7500, 8500, 9500, 10500, 11500, 12500]
+income = sim.calc("equiv_household_net_income")
+    </code>
+  </pre>
+</div>
+
+<script>
+function show_code_7() {
+  var x = document.getElementById("code_block_7");
+  if (x.style.display === "none") {
+    x.style.display = "block";
+  } else {
+    x.style.display = "none";
+  }
+}
+</script>
+
+
+<button onclick="show_code_8()">Click to show code</button>
+<div id="code_block_8" style="display: none;">
+  <pre>
+    <code>
+#Income % change for deciles w vs w/o PA
+class PA_reform(Reform):
+    def apply(self):
+        self.neutralize_variable("personal_allowance")
+        
+sim_no_PA = Microsimulation(PA_reform, year=2020)
+
+income_with_PA = sim.calc("equiv_household_net_income") 
+income_no_PA = sim_no_PA.calc("equiv_household_net_income")
+decile_change = []
+for n in range (1, 11):
+    #total income difference for the decile
+    id = income_with_PA.groupby(income.decile_rank()).sum()[n] - income_no_PA.groupby(income.decile_rank()).sum()[n]
+    #average income difference for the decile
+    id2 = id/income_with_PA.groupby(income.decile_rank()).count()[n]
+    #average income difference of the decile as a percent of average income for the decile
+    id3 = id2/income_with_PA.groupby(income.decile_rank()).mean()[n] * 100
+    decile_change.append(id3.round(2))
+
+change = pd.Series(decile_change)
+chart1 = format_fig((px.bar(y=decile_change, labels={"x":"Decile", "y":"Income gain"}).update_layout(
+    title_text='Income increase brought by the personal allowance for households in each income decile',
+    xaxis_title ="Household income decile",
+    yaxis_title ="Change in household net income",
+    showlegend= False,
+    yaxis_ticksuffix="%",
+    xaxis = dict(
+        tickvals = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
+        ticktext = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+    ),
+    
+).update_traces(marker_color='#1976D2')))
+chart1
+    </code>
+  </pre>
+</div>
+
+<script>
+function show_code_8() {
+  var x = document.getElementById("code_block_8");
+  if (x.style.display === "none") {
+    x.style.display = "block";
+  } else {
+    x.style.display = "none";
+  }
+}
+</script>
+
+<div>
+  <script>
+    $(document).ready(function(){
+      $("#graph_graph_8_1").load("{{site.baseurl}}assets/markdown_assets/personal-allowance/graph_8_1.html");
+    });
+  </script>
+</div>
+<div id = "graph_graph_8_1"></div>
+
+<div>
+  <script>
+    $(document).ready(function(){
+      $("#graph_graph_8_2").load("{{site.baseurl}}assets/markdown_assets/personal-allowance/graph_8_2.html");
```

Comment:
title is too long, gets cut off

how about `Effect of personal allowance on household net income by decile`?